### ISSUE TITLE
False Positive on log4j-api-2.14.1.jar

### ISSUE THREAD

**Issue by ghillert** [dependency-check/DependencyCheck#3868]
False positive on library `log4j-api-2.14.1.jar` - reported as cpe:2.3:a:apache:log4j:2.14.1

```
org.apache.logging.log4j:log4j-api:jar:2.14.1
```

**Comment by ghillert** [comment]
Added the following 2 suppressions:

```xml
<suppress>
<notes><![CDATA[
 file name: log4j-to-slf4j-2.14.1.jar
 ]]></notes>
<packageUrl regex="true">^pkg:maven/org\.apache\.logging\.log4j/log4j\-to\-slf4j@.*$</packageUrl>
<cpe>cpe:/a:apache:log4j</cpe>
</suppress>
<suppress>
<notes><![CDATA[
 file name: log4j-api-2.14.1.jar
 ]]></notes>
<packageUrl regex="true">^pkg:maven/org\.apache\.logging\.log4j/log4j\-api@.*$</packageUrl>
<cpe>cpe:/a:apache:log4j</cpe>
</suppress>
```

See also: https://spring.io/blog/2021/12/10/log4j2-vulnerability-and-spring-boot

**Comment by bergerst** [comment]
As https://github.com/jeremylong/DependencyCheck/pull/3869#issuecomment-998306246 said, 6.5.1 still reports CVE-2021-45046 and CVE-2021-45105.

> One or more dependencies were identified with known vulnerabilities in project:
log4j-api-2.14.1.jar (pkg:maven/org.apache.logging.log4j/log4j-api@2.14.1, cpe:2.3:a:apache:log4j:2.14.1:*:*:*:*:*:*:*) : CVE-2021-45046, CVE-2021-45105
See the dependency-check report for more details.

**Comment by ghillert** [comment]
Can confirm: `log4j-api-2.14.1.jar: CVE-2021-45105, CVE-2021-45046` is still being reported. @jeremylong Can this ticket be re-opened or shall I create a new ticket?

With `6.5.1` I still need to add the following suppression to pass the check:

```xml
    <suppress>
        <notes><![CDATA[
         file name: log4j-api-2.14.1.jar
         ]]></notes>
        <packageUrl regex="true">^pkg:maven/org\.apache\.logging\.log4j/log4j\-api@.*$</packageUrl>
        <cpe>cpe:/a:apache:log4j</cpe>
    </suppress>
```

**Comment by jeremylong** [comment]
This is already fixed in 6.5.2-SNAPSHOT.
